redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1219 @@
+        any valid time length such as `30 minutes` or `2 days`. If nothing
+        is provided the mute will be indefinite.
+
+        Examples:
+        `[p]mutechannel @member1 @member2 spam 5 hours`
+        `[p]mutechannel @member1 3 days`
+        """
+        if not users:
+            return await ctx.send_help()
+        if ctx.me in users:
+            return await ctx.send(_("You cannot mute me."))
+        if ctx.author in users:
+            return await ctx.send(_("You cannot mute yourself."))
+        if not await self._check_for_mute_role(ctx):
+            return
+        async with ctx.typing():
+            duration = time_and_reason.get("duration", {})
+            reason = time_and_reason.get("reason", None)
+            until = None
+            time = ""
+            if duration:
+                until = datetime.now(timezone.utc) + timedelta(**duration)
+                time = _(" for ") + humanize_timedelta(timedelta=timedelta(**duration))
+            else:
+                default_duration = await self.config.guild(ctx.guild).default_time()
+                if default_duration:
+                    until = datetime.now(timezone.utc) + timedelta(**default_duration)
+                    time = _(" for ") + humanize_timedelta(timedelta=timedelta(**default_duration))
+            author = ctx.message.author
+            channel = ctx.message.channel
+            guild = ctx.guild
+            audit_reason = get_audit_reason(author, reason)
+            success_list = []
+            for user in users:
+                success = await self.channel_mute_user(
+                    guild, channel, author, user, until, audit_reason
+                )
+                if success["success"]:
+                    success_list.append(user)
+
+                    try:
+                        await modlog.create_case(
+                            self.bot,
+                            guild,
+                            ctx.message.created_at.replace(tzinfo=timezone.utc),
+                            "cmute",
+                            user,
+                            author,
+                            reason,
+                            until=until,
+                            channel=channel,
+                        )
+                    except RuntimeError as e:
+                        log.error(_("Error creating modlog case"), exc_info=e)
+                else:
+                    return await ctx.send(success["reason"])
+            if success_list:
+                verb = _("has")
+                if len(success_list) > 1:
+                    verb = _("have")
+                await channel.send(
+                    _("{users} {verb} been muted in this channel{time}.").format(
+                        users=humanize_list([f"{u}" for u in success_list]), verb=verb, time=time
+                    )
+                )
+
+    @commands.command(usage="[users...] [reason]")
```

Comment:
Users argument is required.
```suggestion
    @commands.command(usage="<users...> [reason]")
```